libp2p/daemon/daemonapi.nim

```diff
@@ -570,7 +576,8 @@ proc newDaemonApi*(flags: set[P2PDaemonFlags] = {},
                    gossipsubHeartbeatDelay = 0,
                    peersRequired = 2,
                    logFile = "",
-                   logLevel = IpfsLogLevel.Debug): Future[DaemonAPI] {.async.} =
+                   logLevel = IpfsLogLevel.Debug): Future[DaemonAPI]
+                   {.async, raises: [Defect, DaemonLocalError].} =
```

Comment:
actually, this is where things get messy with the current chronos patch: the correct annotation for `async` proc's is actually `raises: [Defect]` - I've described it a bit here: https://github.com/status-im/nim-chronos/pull/166/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R225

effectively, the `eh-tracking` branch doesn't work with "logically correct" raises annotations (yet) - it looks at the raises of the transformed function - this will need more work in the future - for now, just put `push raises: [Defect]` on top and nothing on `async` functions..

Reply:
Hmm, actually it doesn't seem to be the case and if the exception is omitted from the annotation it complains about a leaking generic `Exception`, but  not the concrete type however, after adding the concrete exception type it seems to silence it. It would be good if you try it yourself on that module since I did initially think that all we can work with is `Defect`.

Here is the wording from that link for convenience.

```
### Exception effects

`chronos` currently offers minimal support for exception effects and `raises`
annotations. In general, during the `async` transformation, a generic
`except CatchableError` handler is added around the entire function being
transformed, in order to catch any exceptions and transfer them to the `Future`.
Because of this, the effect system thinks no exceptions are "leaking" because in
fact, exception _handling_ is deferred to when the future is being read.

Effectively, this means that while code can be compiled with
`{.push raises: [Defect]}`, the intended effect propoagation and checking is
**disabled** for `async` functions.
```

Reply:
That's weird, maybe cos its a closure? Did you try extracting a proc?